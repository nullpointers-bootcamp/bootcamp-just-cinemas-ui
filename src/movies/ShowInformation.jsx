import React from "react";
import PropTypes from "prop-types";
import _ from "lodash";
import "./showinformation.css";

class ShowInformation extends React.Component {
  groupShowInformation() {
    const { shows } = this.props;
    return _.groupBy(
      shows,
      item => item.theatreId.toString() + item.screenId.toString()
    );
  }

  render() {
    return _.map(this.groupShowInformation(), (shows, index) => (
      <div key={index} className="show-row">
        <div className="show-theatre-screen">
          <div>{shows[0].theatreName}</div>
          <div>{shows[0].screenName}</div>
        </div>
        <div className="show-timings-wrapper">
          {shows.map((show, index) => (
            <div key={index} className="show-timing">
              {show.time}
            </div>
          ))}
        </div>
      </div>
    ));
  }
}

ShowInformation.defaultProps = {
  shows: []
};

ShowInformation.propTypes = {
  shows: PropTypes.array.isRequired
};

export default ShowInformation;
