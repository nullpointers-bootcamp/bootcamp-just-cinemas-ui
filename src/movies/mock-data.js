export const movieItems = [
  {
    id: 1,
    name: "Kabali",
    experiences: "RDX, Dolby Atmos, SUB",
    listingType: "NOW_SHOWING",
    language: null,
    imageName: "kabali",
    stills: ["kabali_1", "kabali_2"],
    synopsis:
      "A man attempts to protect his family from enemies, gained in his time as a gangster. After he is released from prison on false charges, he resumes his post as gang leader and continues a long-held rivalry."
  },
  {
    id: 2,
    name: "Theri",
    experiences: "RDX, Dolby Atmos, SUB",
    listingType: "NOW_SHOWING",
    language: null,
    imageName: "theri",
    stills: ["theri_1", "theri_2"],
    synopsis:
      "Concealing his identity, DCP Vijaya Kumar goes into self-inflicted exile to bring up his daughter. However, when her life is threatened by gangsters, he must settle an old score with a politician."
  },
  {
    id: 3,
    name: "Banjo",
    experiences: "RDX 3D",
    listingType: "NOW_SHOWING",
    language: null,
    imageName: "banjo",
    stills: ["banjo_1", "banjo_2"],
    synopsis:
      "Taraat, a banjoist hailing from a poor background, and his associates get the offer of a lifetime when they are approached by Christina, a musician from the US."
  },
  {
    id: 4,
    name: "Suicide Squad",
    experiences: "Dolby Atmos, RDX 3D, SUB",
    listingType: "NOW_SHOWING",
    language: null,
    imageName: "suicide-squad",
    stills: ["suicide-squad_1", "suicide-squad_2"],
    synopsis:
      "Figuring they are all expendable, a U.S. intelligence officer decides to assemble a team of dangerous, incarcerated supervillains for a top-secret mission. Now armed with government weapons, Deadshot (Will Smith), Harley Quinn (Margot Robbie), Captain Boomerang, Killer Croc and other despicable inmates must learn to work together. Dubbed Task Force X, the criminals unite to battle a mysterious and powerful entity, while the diabolical Joker (Jared Leto) launches an evil agenda of his own."
  },
  {
    id: 5,
    name: "Namadhu",
    experiences: "RDX, SUB",
    listingType: "NOW_SHOWING",
    language: null,
    imageName: "namadhu",
    stills: ["namadhu_1", "namadhu_2"],
    synopsis:
      "The lives of four middle-class individuals, Sairam, Gayathri, Abhiram and Mahitha, collide as they pursue their individual ambitions amidst various challenges."
  },
  {
    id: 6,
    name: "Pelli Chupulu",
    experiences: "RDX, SUB",
    listingType: "NOW_SHOWING",
    language: null,
    imageName: "pelli-chupulu",
    stills: ["pelli-chupulu_1", "pelli-chupulu_2"],
    synopsis:
      "Vijays father, who is fed up with Vijays carefree life, decides to make him marry a girl in hopes that he might become a responsible man. However, meeting an ambitious girl changes his life for the better."
  },
  {
    id: 7,
    name: "Pink",
    experiences: "RDX, SUB",
    listingType: "NOW_SHOWING",
    language: null,
    imageName: "pink",
    stills: ["pink_1", "pink_2"],
    synopsis:
      "After being molested, Minal, along with her friends, tries to file an FIR against a politicians nephew. When the subsequent case gets rigged, Deepak, a retired lawyer, helps them to fight the case."
  },
  {
    id: 8,
    name: "Remo",
    experiences: "Dolby Atmos, RDX, SUB",
    listingType: "NOW_SHOWING",
    language: null,
    imageName: "remo",
    stills: ["remo_1", "remo_2"],
    synopsis:
      "An aspiring actor falls in love with Kavya, a doctor. Upon knowing that she is already engaged, he disguises himself as a female nurse with the aim of breaking her engagement."
  },
  {
    id: 9,
    name: "Inferno",
    experiences: "Dolby Atmos, RDX, SUB",
    listingType: "NOW_SHOWING",
    language: null,
    imageName: "inferno",
    stills: ["inferno_1", "inferno_2"],
    synopsis:
      "Famous symbologist Robert Langdon (Tom Hanks) follows a trail of clues tied to Dante, the great medieval poet. When Langdon wakes up in an Italian hospital with amnesia, he teams up with Sienna Brooks (Felicity Jones), a doctor he hopes will help him recover his memories. Together, they race across Europe and against the clock to stop a madman (Ben Foster) from unleashing a virus that could wipe out half of the worlds population."
  },
  {
    id: 10,
    name: "Iron Man 1",
    experiences: "RDX, Dolby Atmos, SUB",
    listingType: "NOW_SHOWING",
    language: null,
    imageName: "ironman1",
    stills: ["ironman1_1", " ironman1_2"],
    synopsis:
      "When an industrialist is captured, he constructs a high-tech armoured suit to escape. Once he manages to escape, he decides to use his suit to fight against evil forces and save the world."
  },
  {
    id: 11,
    name: "Iron Man 2",
    experiences: "RDX, Dolby Atmos, SUB",
    listingType: "NOW_SHOWING",
    language: null,
    imageName: "ironman2",
    stills: ["ironman2_2", " ironman2_2"],
    synopsis:
      "Tony Stark is under pressure from various sources, including the government, to share his technology with the world. He must find a way to fight them while also tackling his other enemies."
  },
  {
    id: 12,
    name: "Iron Man 3",
    experiences: "RDX, Dolby Atmos, SUB",
    listingType: "NOW_SHOWING",
    language: null,
    imageName: "ironman3",
    stills: ["ironman3_1", " ironman3_2"],
    synopsis:
      "Tony Stark encounters a formidable foe called the Mandarin. After failing to defeat his enemy, he embarks on a journey with relentless consequences and a suit that ceases to exist."
  },
  {
    id: 13,
    name: "Avengers",
    experiences: "RDX, Dolby Atmos, SUB",
    listingType: "NOW_SHOWING",
    language: null,
    imageName: "avengers",
    stills: ["avengers_1", " avengers_2"],
    synopsis:
      "In the film, Nick Fury, director of the spy agency S.H.I.E.L.D., recruits Tony Stark, Steve Rogers, Bruce Banner, and Thor to form a team that must stop Thors brother Loki from subjugating Earth."
  },
  {
    id: 14,
    name: "Thor",
    experiences: "RDX, Dolby Atmos, SUB",
    listingType: "NOW_SHOWING",
    language: null,
    imageName: "thor",
    stills: ["thor_1", " thor_2"],
    synopsis:
      "Thor is exiled by his father Odin, the King of Asgard, to the Earth to live among mortals. When he lands on Earth, his trusted weapon Mjolnir is discovered and captured by S.H.I.E.L.D."
  },
  {
    id: 15,
    name: "Age of Ultron",
    experiences: "RDX, Dolby Atmos, SUB",
    listingType: "NOW_SHOWING",
    language: null,
    imageName: "ageofultron",
    stills: ["ageofultron_1", " ageofultron_2"],
    synopsis:
      "Tony Stark builds an artificial intelligence system named Ultron with the help of Bruce Banner. When the sentient Ultron makes plans to wipe out the human race, the Avengers set out to stop him."
  }
];

export const movieDetail = {
  id: 1,
  name: "Kabali",
  experiences: "RDX, Dolby Atmos, SUB",
  listingType: "NOW_SHOWING",
  language: null,
  imageName: "kabali",
  stills: ["kabali_1", "kabali_2"],
  synopsis:
    "A man attempts to protect his family from enemies, gained in his time as a gangster. After he is released from prison on false charges, he resumes his post as gang leader and continues a long-held rivalry."
};

export const showInformation = [
  {
    showId: 1,
    date: "2019-02-14",
    time: "09:00",
    theatreId: 1,
    theatreName: "PVR",
    screenId: 1,
    screenName: "Screen 1"
  }
];

export const seatInformation = {
  noOfRows: 2,
  noOfColumns: 3,
  bookedSeats: ["A1"]
};

export const ticketInformaton = {
  bookingRefNumber: "439eab61-08fb-4d4c-96f6-7333c5be7320",
  confirmedSeats: ["H3", "J3", "I3"]
};

export const languages = [];
