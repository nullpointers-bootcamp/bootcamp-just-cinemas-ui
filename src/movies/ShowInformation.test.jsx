import React from "react";
import renderer from "react-test-renderer";
import ShowInformation from "./ShowInformation";
import { showInformation } from "./mock-data";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

it("show show information", () => {
  const tree = renderer
    .create(
      <ShowInformation
        shows={showInformation}
        fetchSeatInformation={() => {}}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("should group show information by theatre-id and screen-id", () => {
  const wrapper = shallow(
    <ShowInformation shows={showInformation} fetchSeatInformation={() => {}} />
  );
  expect(wrapper.instance().groupShowInformation()).toEqual({
    "11": showInformation
  });
});
