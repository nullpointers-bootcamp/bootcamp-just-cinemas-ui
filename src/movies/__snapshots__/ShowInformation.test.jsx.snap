// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`show show information 1`] = `
<div
  className="show-row"
>
  <div
    className="show-theatre-screen"
  >
    <div>
      PVR
    </div>
    <div>
      Screen 1
    </div>
  </div>
  <div
    className="show-timings-wrapper"
  >
    <div
      className="show-timing"
    >
      09:00
    </div>
  </div>
</div>
`;
